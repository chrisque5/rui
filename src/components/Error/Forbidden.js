import { Link, useLocation } from 'react-router-dom';

import logo from '../../img/tpicap.png';

const Forbidden = () => {
  const location = useLocation();

  return (
    <div className="error-page">
      <div className="flex-container">
        <div className="flex-item">
          <img src={logo} alt="Logo" className="logo" />

          <h3 className="title">Deal Management System</h3>
          <h3 className="subtitle">Insufficient Permissions</h3>
          <p>
            Insufficient permissions to access
            &quot;
            {location.pathname}
            &quot;.
          </p>
          <Link to="/" className="linkbutton">Go to Home</Link>
        </div>
      </div>
    </div>
  );
};

export default Forbidden;
