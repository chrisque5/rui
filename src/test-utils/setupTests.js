// The below part of the code is to fix the issue
// TypeError: window.matchMedia is not a function
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});
