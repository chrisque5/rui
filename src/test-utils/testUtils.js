/* eslint-disable import/no-extraneous-dependencies, react/prop-types */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Form } from 'antd';

const middlewares = [thunk];
const renderWithForm = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}><Form>{children}</Form></Provider>;

  return render(ui, { wrapper, ...renderOptions });
};

const renderWithProvider = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}>{children}</Provider>;

  return render(ui, { wrapper, ...renderOptions });
};
// re-export everything
export * from '@testing-library/react';
// override render method
export { renderWithForm, renderWithProvider };
