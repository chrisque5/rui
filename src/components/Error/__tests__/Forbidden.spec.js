import { HashRouter } from 'react-router-dom';
import { render } from 'test-utils/testUtils';
import Forbidden from '../Forbidden';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/forbidden/path',
  }),
  useHistory: () => ({
    goBack: jest.fn(),
  }),
}));

describe('<Forbidden>', () => {
  test('should match the snapshot', () => {
    const { container } = render(<HashRouter><Forbidden /></HashRouter>);
    expect(container).toMatchSnapshot();
  });
});
