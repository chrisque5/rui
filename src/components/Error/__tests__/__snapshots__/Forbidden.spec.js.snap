// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Forbidden> should match the snapshot 1`] = `
<div>
  <div
    class="error-page"
  >
    <div
      class="flex-container"
    >
      <div
        class="flex-item"
      >
        <img
          alt="Logo"
          class="logo"
          src="tpicap.png"
        />
        <h3
          class="title"
        >
          Deal Management System
        </h3>
        <h3
          class="subtitle"
        >
          Insufficient Permissions
        </h3>
        <p>
          Insufficient permissions to access "
          localhost:3000/forbidden/path
          ".
        </p>
        <a
          class="linkbutton"
          href="#/"
        >
          Go to Home
        </a>
      </div>
    </div>
  </div>
</div>
`;
